session.go

```diff
@@ -561,7 +565,7 @@ func (s *session) DropPreparedStmt(stmtID uint32) error {
 	if _, ok := vars.PreparedStmts[stmtID]; !ok {
 		return executor.ErrStmtNotFound
 	}
-	delete(vars.PreparedStmts, stmtID)
+	vars.RetryInfo.DroppedPreparedStmtIDs = append(vars.RetryInfo.DroppedPreparedStmtIDs, stmtID)
```

Comment:
We only retry execute statement.
`Session.PrepareStmt` do not add statement history.